### ISSUE TITLE
Corrupt configuration file: Unknown Ganeti uid '512'

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#283]
Originally reported of Google Code with ID 227.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster (ganeti v2.4.5) 2.4.5


<b>What distribution are you using?</b>

Debian Squeeze, with ganeti2 from backports
ganeti2   2.4.5-1~bpo60+1

<b>What steps will reproduce the problem?</b>
1. trying to add a new node to a cluster with : gnt-node add vsm002

<b>What is the expected output? What do you see instead?</b>

No error expected, or error with real informations. But I see :

# gnt-node add vsm002
-- WARNING --
Performing this operation is going to replace the ssh daemon keypair
on the target machine (vsm002) with the ones of the current one
and grant full intra-cluster ssh root access to/from it

/usr/lib/python2.6/dist-packages/Crypto/Util/randpool.py:40:
RandomPool_DeprecationWarning: This application uses RandomPool, which
is BROKEN in older releases.  See
http://www.pycrypto.org/randpool-broken
 RandomPool_DeprecationWarning)
Unable to verify hostkey of host vsm002:
d6:87:ec:2f:46:11:27:09:09:dc:ee:47:b7:db:60:24. Do you want to accept
it?
y/[n]/?: y
Tue Apr 10 07:16:01 2012  - INFO: Node will be a master candidate
Corrupt configuration file: Unknown Ganeti uid '512'
Aborting.

<b>Please provide any additional information below.</b>

So, the "Corrupt configuration file: Unknown Ganeti uid '512'
Aborting." is not really helpful.
After asking mailing list, iustin asked me to find files owned by uid 512, and we found the problem :

# find / -uid 512
/etc/xen/xend-config.sxp

So i changed ownership of this file to root again, and gnt-node add worked.

It is unclear why this unused user had ownership of the file.
But ganeti should at least give more details according to iustin
```
Originally added on 2012-04-11 07:11:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Just a description of what's happening, not yet a fix:

- RPC uses stat(2) on the file and sees UID 512
- Destination node calls backend.UploadFile
- That function verifies the UID and fails

The source/master node should already verify the UID, but something in your setup and/or in the code did not work correctly. We'll have to investigate this further.
```
Originally added on 2012-04-11 18:43:40 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-07-19 14:57:37 +0000 UTC.

Added Labels: Type-Defect Component-UI

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-08 14:23:11 +0000 UTC.

Added to Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
Move non-critical bugs scheduled for 2.8 or 2.9 to 2.11, as in those versions only critical bug fixes will be integrated.
```
Originally added on 2013-10-30 09:48:16 +0000 UTC.

Added to Milestone: Release2.11
Removed from Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-02-12 15:33:44 +0000 UTC.

Added Labels: Priority-Low

**Comment by Ganeti-Issues-Migrator** [comment]
```
Moving all issues that won't be addressed in 2.11 anymore to 2.12.
```
Originally added on 2014-05-14 11:28:36 +0000 UTC.

Added to Milestone: Release2.12
Removed from Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
I am closing this issue because it's old and it hasn't been updated in a while. This is according to our current team effort to migrate to Github issues. Please feel free to comment and re-open this if you think it's still relevant.

See https://groups.google.com/forum/#!topic/ganeti/b2-eukpsXpk for more details.
```
Originally added on 2016-12-15 10:35:02 +0000 UTC.

Changed State: Obsolete
Removed from Milestone: Release2.12
